app/presentation/RoomItem/Touchable.tsx

```diff
@@ -28,255 +34,214 @@ interface ITouchableProps {
 	displayMode: string;
 }
 
-class Touchable extends React.Component<ITouchableProps, any> {
-	private dragX: Animated.Value;
+const Touchable = (props: ITouchableProps): JSX.Element => {
+	const { testID, isRead, width, favorite, children, theme, isFocused, swipeEnabled, displayMode } = props;
 
-	private rowOffSet: Animated.Value;
+	const rowOffSet = useSharedValue(0);
+	const transX = useSharedValue(0);
+	const rowState = useSharedValue(0); // 0: closed, 1: right opened, -1: left opened
+	let _value = 0;
 
-	private reverse: Animated.Value;
+	const close = () => {
+		rowState.value = 0;
+		transX.value = withSpring(0, { overshootClamping: true });
+		rowOffSet.value = 0;
+	};
 
-	private transX: Animated.AnimatedAddition;
+	const toggleFav = () => {
+		const { toggleFav, rid, favorite } = props;
```

Comment:
Since we will destructure everything directly on the  component, we don't need to destructure `props` here anymore